pkg/trace/agent/obfuscate.go

```diff
@@ -126,6 +126,7 @@ func newCreditCardsObfuscator(cfg config.CreditCardsConfig) *ccObfuscator {
 	if cfg.Enabled {
 		// obfuscator disabled
 		pb.SetMetaHook(cco.MetaHook)
+		pb.SetMetaStructHook(cco.MetaStructHook)
```

Comment:
When looking at the unchanged code, we can see this got integrated into the existing meta hook API. I would go 100% into this choice hiding `pb.SetMetaStructHook()` and changing `pb.SetMetaHook()` so that it takes the meta struct hook as second argument.

I was also concerned by the following line of code which only tests the presence of a the regular meta hook to run both of them: https://github.com/DataDog/datadog-agent/blob/a58e1c4a5a770bed7bc75339b9a9163d4fed3e58/pkg/trace/api/api.go#L584